code/game/machinery/_machinery.dm

```diff
@@ -541,3 +549,117 @@ Class Procs:
 	. = . % 9
 	AM.pixel_x = -8 + ((.%3)*8)
 	AM.pixel_y = -8 + (round( . / 3)*8)
+
+/obj/machinery/proc/id_eject_prisoner(mob/user)
+	if(inserted_prisoner_id)
+		inserted_prisoner_id.forceMove(drop_location())
+		if(!issilicon(user) && Adjacent(user))
+			user.put_in_hands(inserted_prisoner_id)
+			inserted_prisoner_id = null
+			user.visible_message("<span class='notice'>[user] gets an ID card from the console.</span>", \
+								"<span class='notice'>You get the ID card from the console.</span>")
+			playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+		updateUsrDialog()
+		return
+	else
+		to_chat(user, "<span class='warning'>There's no ID card in the console!</span>")
+
+	if(!inserted_prisoner_id)
+		to_chat(user, "<span class='warning'>There's no ID card in the console!</span>")
+		return
+	if(inserted_prisoner_id)
+		inserted_prisoner_id.forceMove(drop_location())
+		if(!issilicon(user) && Adjacent(user))
+			user.put_in_hands(inserted_prisoner_id)
+		inserted_prisoner_id = null
+		user.visible_message("<span class='notice'>[user] gets an ID card from the console.</span>", \
+							"<span class='notice'>You get the ID card from the console.</span>")
+		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+		updateUsrDialog()
+
+/obj/machinery/proc/id_insert_prisoner(mob/user)
+	if(inserted_prisoner_id)
+		to_chat(user, "<span class='warning'>There's already an ID card in the console!</span>")
+		return
+	var/obj/item/card/id/prisoner/I = user.get_active_held_item()
+	if(istype(I))
+		if(!user.transferItemToLoc(I, src))
+			return
+		inserted_prisoner_id = I
+		user.visible_message("<span class='notice'>[user] inserts an ID card into the console.</span>", \
+							"<span class='notice'>You insert the ID card into the console.</span>")
+		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+	else
+		to_chat(user, "<span class='danger'>No valid ID.</span>")
+	updateUsrDialog()
+
+/obj/machinery/proc/id_insert_scan(mob/user, obj/item/card/id/I)
+	I = user.get_active_held_item()
```

Comment:
Not what I meant but I got a good laugh from this. In the attack_by that's calling this, cast the attacking item to an ID card, then pass it it into the id_insert proc.

Reply:
Actually I don't think you need the cast, but to clarify what you're doing here is passing something in a parameter and then immediately overwriting it with an assignment defeating the whole point of the pass.